install/0000_30_machine-api-operator_10_service.yaml

```diff
@@ -18,3 +18,32 @@ spec:
   selector:
     k8s-app: machine-api-operator
   sessionAffinity: None
+---
+apiVersion: v1
```

Comment:
where is the serviceMonitor that goes with this?

Reply:
It can’t be merged until provider integration PRs get into repos, or the CI would fail as the machine metrics won’t be accessible. The serviceMonitor i therefore in https://github.com/openshift/machine-api-operator/pull/609

Reply:
The provider PRs are 
- https://github.com/openshift/cluster-api-provider-gcp/pull/94
- https://github.com/openshift/cluster-api-provider-azure/pull/136
- https://github.com/openshift/cluster-api-provider-aws/pull/324

Reply:
>or the CI would fail as the machine metrics won’t be accessible

can you elaborate? can you point me to the origin test that will fail?

Reply:
Have a look into ´Alerts shouldn't report any alerts in firing state...´ test in every provider: sample https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_machine-api-operator/590/pull-ci-openshift-machine-api-operator-master-e2e-aws/2496 


https://coreos.slack.com/archives/GE2HQ9QP4/p1591280443332000?thread_ts=1591279283.331700&channel=GE2HQ9QP4&message_ts=1591280443.332000

Reply:
why would this fire an alert?

Reply:
It's not the service which is triggering it, but the serviceMonitor from the second PR https://github.com/openshift/machine-api-operator/pull/609 I included the merge order in Jira issue, posted it in slack too. While provider PRs are not merged, the metrics port is not served from the code, and while Prometheus is trying to connect, it machine metrics responds with 502. That causes an alert in openshift-monitoring namespace. Joel and me already discussed the issue in slack, and decided to split the PR, instead of revendoring MAO PR branch in every provider.

Reply:
I thought prometheus was no triggering alert for that scenario. This would need to account for openstack and baremetal.

Reply:
I’ve opened issues in both repos to make sure this won’t be left unanswered.
- https://github.com/openshift/cluster-api-provider-openstack/issues/100
- https://github.com/openshift/cluster-api-provider-baremetal/issues/75

including machineset and mhc in the metrics should not be disruptive.